### PR TITLE
Create calendar

### PR DESCRIPTION
**Pull request by stevehackreactor**
Set up calendar creation when clients log into our service for the first time.
This is done shortly after we instantiate our client in our postgres database.

### CODE REVIEW

File: api/src/passport.setup.js [hunk 1]
```diff
@@ -42,11 +38,23 @@ passport.use(
...
-      const token = jwt.sign({ accessToken, refreshToken, googleKey: profile.id }, process.env.JWT_SECRET);
-      return done(null, token);
+        }
+      })
+        .then((user) => {
+          if (!user[0].calendarId) {
+            console.log('making calendar for: ', user[0]);
+            let calendar = Calendar.createCalendar(accessToken, refreshToken);
+            Promise.resolve(calendar)
+              .then((calendarObject) => {
```
Comment (chaines) [inline api/src/passport.setup.js:47]: While this definitely works, you should be able to just do: 
```
Calendar.createCalendar(accessToken, refreshToken)
.then((calendarObject) => {
//...
```
I'm curious if there's a specific reason you didn't do it this way.

File: api/src/passport.setup.js [hunk 2]
```diff
@@ -42,11 +38,23 @@ passport.use(
...
+                return user[0].save();
+              });
+          }
+        })
+        .then(()=> {
+          const token = jwt.sign({ accessToken, refreshToken, googleKey: profile.id }, process.env.JWT_SECRET);
+          return done(null, token);
+        });
     }
   )
```
Comment (chaines) [inline api/src/passport.setup.js:61]: Make sure to add a 'catch' block here, that calls done with the error.

File: api/src/passport.setup.js (hunk 1 above)
Comment (stevehackreactor) [reply api/src/passport.setup.js:47]: I have refactored this code to save those few lines.

File: api/src/passport.setup.js (hunk 2 above)
Comment (stevehackreactor) [reply api/src/passport.setup.js:61]: I have now added the catch block.

**Pull request merged**